[Doc-only] Update descriptor README and x-descriptors guide (#5530)
openshift/console: frontend/packages/operator-lifecycle-manager/src/components/descriptors/README.md

![screenshot_20180821_163304](https://user-images.githubusercontent.com/11700385/44427562-eb1fb500-a55f-11e8-83e5-98e7008dabdb.png)

## Create Operand Form

> In Kubernetes 1.8, CRDs gained the ability to define an optional [OpenAPI v3 based validation schema](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#schemaObject). This is important for data consistency, security, and can also be used to auto-generate the creation form for the instances of CRD (or "Operands").

Reviewer: ```suggestion
In Kubernetes 1.8, CRDs gained the ability to define an optional [OpenAPI v3 based validation schema](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#schemaObject). This is important for data consistency, security, and can also be used to auto-generate forms for creating CRD instances (or "Operands").
```

### Auto-generated Form

> With a [structural schema](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#specifying-a-structural-schema) (mandatory for CRD in `apiextensions.k8s.io/v1`), OpenShift Console can generate a Creation Form for the Operands automatically using [react-jsonschema-form](https://rjsf-team.github.io/react-jsonschema-form/) to map out the form fields based on the property types, e.g. `boolean`, `text`, `integer`, `string`, `string` with `enum`, `object`, and `array` with associate UI components.

Reviewer: ```suggestion
With a [structural schema](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#specifying-a-structural-schema) (mandatory for CRD in `apiextensions.k8s.io/v1`), OpenShift Console can automatically generate creation forms for operands. Schema properties are mapped to appropriate form field UI components using [react-jsonschema-form](https://rjsf-team.github.io/react-jsonschema-form/). These mappings are primarily based on schema types (e.g. `boolean`, `text`, `integer`, `string`, `string` with `enum`, `object`, and `array`).
```

### Customized Creation Form with specDescriptors

> The auto-generated creation form is already powerful but you may want to customize how the creation form is rendered. For example, a more user-friendly `displayName`, a more concise `description` for the form fields, change the [form field ordering](#ordering-of-form-fields), or simply want to [exclude some non-user facing fields](reference/reference.md#21-hidden) from the creation form. This is where "specDescriptors" come in.

Reviewer: ```suggestion
The auto-generated creation form is already powerful, but you may want to customize how the form is rendered. For example, you may want a more user-friendly `displayName`, a more concise `description`, to change [form field ordering](#ordering-of-form-fields), or to [exclude some non-user facing fields](reference/reference.md#21-hidden). This is where "specDescriptors" come in.
```

> Checkout the [reference](reference/reference.md) of different `x-descriptors` for the specDescriptors (and statusDescriptors) available with sample code snippets and screenshots.

Reviewer: ```suggestion
Check out the `x-descriptors` [reference guide](reference/reference.md) which provides more detailed descriptions and examples of how to use them in specDescriptors and statusDescriptors.
```

### Ordering of Form Fields

> The ordering of form fields is depending on whether a field has a specDescriptor, and is "required" (a `required:` property in the schema), "optional", or "advanced" (assigned with an [advanced](reference/reference.md#18-advanced) specDecriptor). In general, a field with a specDescriptor has a higher sort-weight comparing to those without; a required field has a higher sort-weight number comparing to an optional field, as the following set of criteria:

Reviewer: ```suggestion
The ordering of form fields depends on whether a field has a specDescriptor, is "required" (a `required:` property in the schema), "optional", or "advanced" (assigned with an [advanced](reference/reference.md#18-advanced) specDecriptor). In general, fields with specDescriptors will be sorted higher than those without and required fields will be sorted higher than optional fields. The full set of sorting rules are as follows:
```
